test/functional/change_stream.test.js

```diff
@@ -1924,7 +1924,10 @@ describe('Change Streams', function () {
     });
 
     it('when invoked using eventEmitter API', {
-      metadata: { requires: { topology: 'replicaset', mongodb: '>=3.6' } },
+      metadata: {
+        requires: { topology: 'replicaset', mongodb: '>=3.6' },
+        sessions: { skipLeakTests: true }
```

Comment:
Note, this test was leaking a session for me locally. Looks like this is because the client can close in the `afterEach` hook before the `changeStream.close` completes. This seems acceptable, so I skipped the leak checker on this test.

Reply:
why is it:
```
if (counter === 2) {
  changeStream.close();
  setTimeout(() => close());
  ...
}
```
instead of:
```
if (counter === 2) {
  changeStream.close(close);
  ...
}
```
It seems problematic to use `setTimeout` in these cases, it can lead to leaks as you've seen here.

Reply:
Good question, fixed 👍